hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -706,7 +706,20 @@ protected void compactIfNecessary(AbstractHoodieWriteClient writeClient, String
     }
   }
 
-  protected void doClean(AbstractHoodieWriteClient writeClient, String instantTime) {
+  protected void cleanIfNecessary(AbstractHoodieWriteClient writeClient, String instantTime) {
+    Option<HoodieInstant> lastCompletedCompactionInstant = metadataMetaClient.reloadActiveTimeline()
```

Comment:
is reloadActiveTimeline() neceassary here?

Reply:
Also, can you check if there is already a metadata table function to get the last compaction timestamp? 

I guess there are other code paths where this is required. So would be a good idea to create a utility function if does not exist.

Reply:
> is reloadActiveTimeline() neceassary here?

Yes, it is necessary, because we can not make sure whether the compaction triggers successfully or not.

Reply:
Agree, where should i put the util method ?

Reply:
reloadActiveTimeline() seems to be called right before calling this function. Can you please check?

There are utility methods in HoodieTableMetadataUtil class. Also, check if HoodieTableMetadata already has this function.

Reply:
> reloadActiveTimeline() seems to be called right before calling this function

Don't think so, we should still reload the timeline if there are inline compaction triggers and completes.

> check if HoodieTableMetadata already has this function

`HoodieTableMetadata` has this function but just like it name says, it is a metadata reader while the logic is in writer now.